mexico-city/src/mc_nitro.rs

```diff
@@ -0,0 +1,219 @@
+//! AWS Nitro-Enclaves-specific material for the Mexico City enclave
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use byteorder::{ByteOrder, LittleEndian};
+use nix::sys::socket::listen as listen_vsock;
+use nix::sys::socket::{accept, bind, recv, send, MsgFlags, SockAddr};
+use nix::sys::socket::{socket, AddressFamily, SockFlag, SockType};
+use std::os::unix::io::RawFd;
+use veracruz_utils::{MCMessage, vsocket, NitroRootEnclaveMessage, NitroStatus, receive_buffer, send_buffer};
+use std::os::unix::io::AsRawFd;
+use std::convert::TryInto;
+use nsm_io;
+use nsm_lib;
+
+const CID: u32 = 0xFFFFFFFF; // VMADDR_CID_ANY
+const HOST_CID: u32 = 3; // This may not be right
+const PORT: u32 = 5005;
+// max number of outstanding connectiosn in the socket listen queue
+const BACKLOG: usize = 128;
+const OCALL_PORT: u32 = 5006;
+use crate::managers;
+
+// the following value was copied from https://github.com/aws/aws-nitro-enclaves-sdk-c/blob/main/source/attestation.c
+// I've no idea where it came from (I've seen no documentation on this), but
+// I guess I have to trust Amazon on this one
+const NSM_MAX_ATTESTATION_DOC_SIZE: usize = 16 * 1024;
+
+pub fn nitro_main() -> Result<(), String> {
```

Comment:
Following the `rustdoc` comment above, can comments also be added to all of the (public, at least) functions exported from a module?

Also, I think in general we've been moving away from using `String` as the error handling mechanism in favour of dedicated error types with e.g. `Display` implementations for proper pretty-printing of errors (e.g. `durango/src/error.rs`).  Can the failure modes of this function, and the others within this file, be extracted out in a dedicated type in that way?  That approach allows you to look at the error type definition and see (an upper bound on) all the ways that the function could possibly fail immediately.